dashboard/api_edx_cache_test.py

```diff
@@ -0,0 +1,306 @@
+        cls.edx_client.certificates.get_student_certificates.return_value = cls.certificates
+        cls.edx_client.current_grades.get_student_current_grades.return_value = cls.current_grades
+
+    def assert_cache_in_db(self, enrollment_keys=None, certificate_keys=None, grades_keys=None):
+        """
+        Helper function to assert the course keys in the database cache
+        """
+        enrollment_keys = enrollment_keys or []
+        certificate_keys = certificate_keys or []
+        grades_keys = grades_keys or []
+        enrollments = CachedEdxUserData.get_cached_edx_data(self.user, CachedEdxUserData.ENROLLMENT)
+        certificates = CachedEdxUserData.get_cached_edx_data(self.user, CachedEdxUserData.CERTIFICATE)
+        grades = CachedEdxUserData.get_cached_edx_data(self.user, CachedEdxUserData.CURRENT_GRADE)
+        assert sorted(list(enrollments.enrollments.keys())) == sorted(enrollment_keys)
+        assert sorted(list(certificates.certificates.keys())) == sorted(certificate_keys)
+        assert sorted(list(grades.current_grades.keys())) == sorted(grades_keys)
+
+    def test_constants(self):
+        """Tests class constants"""
+        assert CachedEdxUserData.SUPPORTED_CACHES == (
+            CachedEdxUserData.ENROLLMENT,
+            CachedEdxUserData.CERTIFICATE,
+            CachedEdxUserData.CURRENT_GRADE,
+        )
+        assert CachedEdxUserData.CACHED_EDX_MODELS == {
+            CachedEdxUserData.ENROLLMENT: models.CachedEnrollment,
+            CachedEdxUserData.CERTIFICATE: models.CachedCertificate,
+            CachedEdxUserData.CURRENT_GRADE: models.CachedCurrentGrade,
+        }
+        assert CachedEdxUserData.CACHE_EXPIRATION_DELTAS == {
+            CachedEdxUserData.ENROLLMENT:  timedelta(minutes=5),
+            CachedEdxUserData.CERTIFICATE: timedelta(hours=6),
+            CachedEdxUserData.CURRENT_GRADE: timedelta(hours=1),
+        }
+
+    def test_get_cached_edx_data(self):
+        """
+        Test for get_cached_edx_data
+        """
+        with self.assertRaises(ValueError):
+            CachedEdxUserData.get_cached_edx_data(self.user, 'footype')
+
+        self.assert_cache_in_db()
+        for run in self.all_runs:
+            CachedEnrollmentFactory.create(user=self.user, course_run=run)
+            CachedCertificateFactory.create(user=self.user, course_run=run)
+            CachedCurrentGradeFactory.create(user=self.user, course_run=run)
+        self.assert_cache_in_db(self.all_course_run_ids, self.all_course_run_ids, self.all_course_run_ids)
+
+    def test_update_cache_last_access(self):
+        """Test for update_cache_last_access"""
+        with self.assertRaises(ValueError):
+            CachedEdxUserData.update_cache_last_access(self.user, 'footype')
+        assert UserCacheRefreshTime.objects.filter(user=self.user).exists() is False
+
+        CachedEdxUserData.update_cache_last_access(self.user, CachedEdxUserData.ENROLLMENT)
```

Comment:
this might be hard-ish
the tests use data accessible via `self` that is not accessible in the decorator.
Plus the certificates test is slightly different.